From a54d970d74510e7a4e95377ed4d5ef1f9ae1aabc Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 1 Dec 2017 11:10:29 +0800
Subject: [PATCH 3010/5242] MLK-16976-2 ARM64: dts: fsl-imx8qm: let USB have
 wakeup capability

commit  f19a403482a7e0f019c8f2243631b6046f70c033 from
https://source.codeaurora.org/external/imx/linux-imx.git

Let PM code know USB has wake system up capability.

BuildInfo:
- SCFW 245582b, IMX-MKIMAGE 0ad6069a, ATF 6bd98a3
- U-Boot 2017.03-imx_v2017.03+gfa65b0a

Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index b09f2b7..96ef9fd 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -411,12 +411,14 @@
 				reg = <SC_R_USB_0>;
 				#power-domain-cells = <0>;
 				power-domains = <&pd_conn>;
+				wakeup-irq = <267>;
 			};
 
 			pd_conn_usbotg0_phy: PD_CONN_USB_0_PHY {
 				reg = <SC_R_USB_0_PHY>;
 				#power-domain-cells = <0>;
 				power-domains = <&pd_conn>;
+				wakeup-irq = <267>;
 			};
 
 			pd_conn_usbotg1: PD_CONN_USB_1 {
@@ -428,11 +430,13 @@
 				reg = <SC_R_USB_2>;
 				#power-domain-cells = <0>;
 				power-domains = <&pd_conn>;
+				wakeup-irq = <271>;
 			};
 			pd_conn_usb2_phy: PD_CONN_USB_2_PHY {
 				reg = <SC_R_USB_2_PHY>;
 				#power-domain-cells = <0>;
 				power-domains = <&pd_conn>;
+				wakeup-irq = <271>;
 			};
 			pd_conn_sdch0: PD_CONN_SDHC_0 {
 				reg = <SC_R_SDHC_0>;
@@ -2667,6 +2671,7 @@
 	usbotg1: usb@5b0d0000 {
 		compatible = "fsl,imx7d-usb", "fsl,imx27-usb";
 		reg = <0x0 0x5b0d0000 0x0 0x200>;
+		interrupt-parent = <&wu>;
 		interrupts = <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>;
 		fsl,usbphy = <&usbphy1>;
 		fsl,usbmisc = <&usbmisc1 0>;
@@ -2684,7 +2689,7 @@
 			<0x0 0x5B140000 0x0 0x10000>,
 			<0x0 0x5B160000 0x0 0x40000>,
 			<0x0 0x5B120000 0x0 0x10000>;
-		interrupt-parent = <&gic>;
+		interrupt-parent = <&wu>;
 		interrupts = <GIC_SPI 271 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QM_USB3_LPM_CLK>,
 			<&clk IMX8QM_USB3_BUS_CLK>,
-- 
1.7.9.5

